File: SDKs/googleapis.com/dfareporting/v3.3/python/src/sdk/models/shared/activities.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class Activities:
    r"""Activities
    Represents an activity group.
    """
    
    filters: Optional[List[DimensionValue]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filters') }})
    kind: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('kind') }})
    metric_names: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('metricNames') }})
    

File: SDKs/googleapis.com/dfareporting/v3.4/typescript/src/sdk/models/shared/activities.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
import { DimensionValue } from "./dimensionvalue";


// Activities
/** 
 * Represents an activity group.
**/
export class Activities extends SpeakeasyBase {
  @Metadata({ data: "json, name=filters", elemType: shared.DimensionValue })
  filters?: DimensionValue[];

  @Metadata({ data: "json, name=kind" })
  kind?: string;

  @Metadata({ data: "json, name=metricNames" })
  metricNames?: string[];
}
